ai/src/main/java/org/triplea/ai/flowfield/FlowFieldAi.java

```diff
@@ -3,41 +3,174 @@
 import games.strategy.engine.data.GameData;
 import games.strategy.engine.data.GamePlayer;
 import games.strategy.engine.data.GameState;
+import games.strategy.engine.data.Resource;
 import games.strategy.engine.framework.startup.ui.PlayerTypes;
+import games.strategy.engine.player.IPlayerBridge;
+import games.strategy.triplea.Constants;
 import games.strategy.triplea.ai.AbstractAi;
 import games.strategy.triplea.delegate.remote.IAbstractPlaceDelegate;
 import games.strategy.triplea.delegate.remote.IMoveDelegate;
 import games.strategy.triplea.delegate.remote.IPurchaseDelegate;
 import games.strategy.triplea.delegate.remote.ITechDelegate;
+import games.strategy.triplea.ui.TripleAFrame;
+import games.strategy.triplea.ui.menubar.DebugMenu;
+import java.awt.event.ActionEvent;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.swing.AbstractAction;
+import javax.swing.ButtonGroup;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import javax.swing.JRadioButtonMenuItem;
+import lombok.Getter;
+import org.triplea.ai.flowfield.map.DiffusionMap;
+import org.triplea.ai.flowfield.map.DiffusionType;
+import org.triplea.ai.flowfield.map.TerritoryDebugUiAction;
+import org.triplea.ai.flowfield.map.offense.EnemyCapitals;
+import org.triplea.ai.flowfield.map.offense.ResourceToGet;
+import org.triplea.ai.flowfield.odds.LanchesterDebugUiAction;
+import org.triplea.swing.SwingAction;
 
 public class FlowFieldAi extends AbstractAi {
+
+  @Getter private final Map<String, DiffusionMap> diffusions = new HashMap<>();
+  private int round = -1;
+
   public FlowFieldAi(final String name, final PlayerTypes.AiType playerType) {
     super(name, playerType);
   }
 
+  @Override
+  public void initialize(final IPlayerBridge playerBridge, final GamePlayer gamePlayer) {
+    super.initialize(playerBridge, gamePlayer);
+    DebugMenu.registerMenuCallback(getName(), this::addDebugMenuItems);
+
+    final DiffusionType enemyCapitals =
```

Comment:
This code of adding the two diffusions is not the final thing.  It is there just to make sure things are building and showing up.